### PR TITLE
test: add tests for a global cache

### PR DESCRIPTION
**Pull request by koba04**
I've refactored tests for errors at #1404 to use a custom cache for each test, and will refactor other tests as well. Before doing that, I've added tests for the global cache not to break the behaviors of the global cache.

### CODE REVIEW

File: test/use-swr-cache.test.tsx [hunk 1]
```diff
@@ -426,3 +386,115 @@ describe('useSWR - cache provider', () => {
...
+      mutate = mutateWithCache
+      const { data } = useSWR(key, null)
+      return <div>data:{data}</div>
+    }
+
+    function App() {
+      return <Page />
+    }
+
+    render(<App />)
```
Comment (huozhi) [inline test/use-swr-cache.test.tsx:448]:
```suggestion
    render(<Page />)
```

File: test/use-swr-cache.test.tsx [hunk 2]
```diff
@@ -426,3 +386,115 @@ describe('useSWR - cache provider', () => {
...
+      const { data } = useSWR(keys[index], fetcher)
+
+      return <div onClick={() => setIndex(1)}>{data}</div>
+    }
+
+    function App() {
+      return <Section />
+    }
+
+    const { container } = render(<App />)
```
Comment (huozhi) [inline test/use-swr-cache.test.tsx:422]:
```suggestion
    const { container } = render(<Section />)
```

File: test/use-swr-cache.test.tsx [hunk 3]
```diff
@@ -426,3 +386,115 @@ describe('useSWR - cache provider', () => {
...
+    function Foo() {
+      const { data } = useSWR(key, () => value++, {
+        dedupingInterval: 0
+      })
+      return <>{String(data)}</>
+    }
+    function Page() {
+      return <Foo />
+    }
+    render(<Page />)
```
Comment (huozhi) [inline test/use-swr-cache.test.tsx:466]: Or just rename Foo -> Page

```suggestion
    render(<Foo />)
```

**Review by huozhi — APPROVED** [review]

**Comment by huozhi** [comment]
Thanks!

**Comment by huozhi** [comment]
@koba04 How you feel about using jest `setupFiles` to reset cache in jest setup configuration? IMO that would be a shorter way to clean cache for each test case if it's possible

**Comment by koba04** [comment]
Thank you!

@huozhi 
Yes, you are right. resetting a global cache in jest's setup file is the easiest way to solve problems caused by sharing a global cache.
Currently, the `Cache` interface doesn't have the `clear` method, so we would have to add it to the interface to do that. 

Personally, I prefer to create a custom cache for each test even though it requires more works because I'd like to avoid using any shared store between tests and SWR now provides the way. As the blog post for v1 mentioned, this is a valid use case of the custom cache feature and that makes tests more reliable than depending on a global cache.
But I don't have a strong opinion for that, so I'm ok that maintainers choose to keep using a global cache.

**Comment by huozhi** [comment]
@koba04 thanks! It was just an idea comming to my head to save efforts, not 100% sure if it's easy and possible. Feel free to choose they way you think make much more sense! 👍

**Comment by koba04** [comment]
@huozhi Thank you for your feedback!

**Pull request merged**